Axisymmetric mesh and targeted symmetry family updates

lamkina opened the pull request:
## Purpose
<!--
Explain the goal of this PR, if it addresses an existing issue be sure to link to it.
Describe the big picture of your changes here, perhaps using a bullet list if multiple changes are done to accomplish a single goal.
If this PR accomplishes multiple goals, it may be best to create separate PR's for each.
-->
This PR adds the CGNS BC indices for axisymmetric and polar topologies. Additionally, this adds the capability to target specific mesh families when setting a symmetric plane nodes to zero.

## Expected time until merged
<!--
Comment on whether or not this change is urgent and how long you expect this PR to take to review and be merged.
For example, a week would be a reasonable time frame for an average, non-urgent PR.
-->
2-3 weeks

## Type of change
<!--
What types of change is it?
Select the appropriate type(s) that describe this PR
-->

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [ ] Maintenance update
- [ ] Other (please describe)

## Testing
<!-- Explain the steps needed to test the new code to verify that it does indeed address the issue and produce the expected behavior. -->

## Checklist
<!-- Put an `x` in the boxes that apply. -->

- [x] I have run `flake8` and `black` to make sure the Python code adheres to PEP-8 and is consistently formatted
- [ ] I have formatted the Fortran code with `fprettify` or C/C++ code with `clang-format` as applicable
- [ ] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

eirikurj commented on cgnsutilities/cgnsutilities.py:
```diff
@@ -1858,7 +1863,7 @@ def extrude(self, direction):
 
         # Get the data order and new dims
-        order, newDims = self._extrudeGetDataOrderAndDIms(direction)
+        order, newDims = self._extrudeGetDataOrderAndDIms(direction, 1)
```
This is conflicting, and step value should be 2.

lamkina replied:
fixed

eirikurj commented on cgnsutilities/cgnsutilities.py:
```diff
@@ -2374,13 +2380,16 @@ def randomize(self, keepRHS):
-                # the low end since it was in fortran 1-based ordering
-                r = bc.ptRange.copy()
-                self.coords[r[0, 0] - 1 : r[0, 1], r[1, 0] - 1 : r[1, 1], r[2, 0] - 1 : r[2, 1], idir] = 0.0
+            if "symmetry" in bc.internalType:
```
There are only two standard CGNS bc types defined that have `symmetry` in the name. I would prefer making this explicit and do a full check and just adding the other options instead of this partial check.

lamkina replied:
Made the check explicit.

eirikurj reviewed (changes requested):
Did a quick skim over this. Let me know if you have questions and we can discuss.

lamkina commented:
Addressed the comments from @eirikurj . I also added two small tests that check if the z-coordinate is actually zero after running the `symmZero` method with and without a target family.
